Resolving encoding issues

lsolodkova opened the pull request:
Will answer #51 and #52.

lsolodkova committed:
catching UnicodeError and binascii.Error

mvondracek commented:
When this PR is ready for review, please change status from *draft* and assign reviewers so we get notification. :)

lsolodkova committed:
Fixing hexa strings case

Catching issues like non-ASCII characters in entropy/seed hex strings and UnicodeDecode (the last raises while reading the file, so should handle it separately)

lsolodkova committed:
syntax fix

lsolodkova committed:
Internal encoding checks

lsolodkova committed:
Unit tests, part 1

lsolodkova committed:
Unit tests, part 2

lsolodkova committed:
Catching non-UTF-8 passwords in mnemoniccli

lsolodkova committed:
Style fix

lsolodkova committed:
Some CLI tests

lsolodkova committed:
Minor style fixes + parameter fix

lsolodkova committed:
Syntax fix

lsolodkova committed:
One more fix

lsolodkova committed:
Removing extra stuff

lsolodkova committed:
More CLI tests

lsolodkova committed:
Style fix

lsolodkova commented:
Can't write a proper test for non-Unicode entropy/seed, exception occurs not while reading file, but only in unhexlify(). Any hints will be appreciated.

lsolodkova committed:
Removing extra code

sobuch commented on pa193mnemonicslytherin/mnemoniccli.py:
```diff
@@ -242,19 +242,32 @@ def action_generate(config: Config) -> ExitCode:
         with open(config.entropy_filepath, config.format.read_mode) as file:
             entropy = file.read()  # type: typing.Union[bytes, str]
-    except FileNotFoundError as e:
+    except (FileNotFoundError, UnicodeError) as e:
```
is there a reason for using this style and not something like the following?
```
try:
    ...
except error1:
   ...
except error2:
   ...
```

lsolodkova replied:
No, I was hoping to reduce the amound of code for a bit, but this does not really matter. What to do with codeclimate issues don't know, I was trying not to modify your code much.

sobuch replied:
if you dont see some quick fixes for the complexity issues, we can probably let it be, I dont think its worth spending that much time on now (>8hrs estimated in codeclimate)

sobuch commented:
please rebase on dev so we can fuzz this a little

lsolodkova committed:
Revert "Removing extra code"

This reverts commit 80b0450aaabb124f29c85bc06ee56d45c077449e.

lsolodkova commented:
Rebased on dev.

lsolodkova commented:
Tests for non-Unicode entropy/seed work properly only if encoding is given explicitly in read() function (but I'm not able to make so cardinal changes to the code), otherwise we fail only inside unhexlify().

sobuch committed:
fuzzer change

lsolodkova committed:
Encoding tests for extract_checksum and _DictionaryAccess

mvondracek commented on pa193mnemonicslytherin/mnemonic.py:
```diff
@@ -253,10 +259,10 @@ def to_seed(self, seed_password: str = '') -> Seed:
         seed_password = normalize('NFKD', seed_password)
         passphrase = "mnemonic" + seed_password
-        passphrase = passphrase.encode()
+        passphrase = passphrase.encode('utf-8')
```
`'utf-8'` is default value, so both version of this code are correct
https://docs.python.org/3/library/stdtypes.html#str.encode

mvondracek committed:
test_mnemonic: extract invalid utf8 input data

mvondracek committed:
test_mnemonic: fix test names

mvondracek committed:
test_mnemonic: cleanup

mvondracek committed:
test_mnemoniccli: fix names

mvondracek committed:
remove already fixed TODO

already tested in `TestMain.test_arguments_error` as:
~~~py
self.assert_program_error([self.SCRIPT, '-p', 'a' *
(MAX_SEED_PASSWORD_LENGTH + 1)], ExitCode.ARGUMENTS)
~~~

mvondracek committed:
fix names

mvondracek committed:
cleanup

mvondracek committed:
mnemoniccli: argument -p must be valid UTF-8

mvondracek committed:
cleanup

mvondracek committed:
fix: error on password with invalid UTF-8

mvondracek committed:
cleanup

mvondracek committed:
fix names

mvondracek committed:
remove solved TODOs

mvondracek committed:
tests: add odd length hex strings for entropy and seed

mvondracek committed:
tests: specify UTF-8 encoding

mvondracek committed:
fix names

mvondracek committed:
test_mnemonic.TestMain.assert_program_cli check stderr first

mvondracek committed:
fix: mnemoniccli: wrong indent for generate, recover, verify

mvondracek committed:
remove solved TODOs

mvondracek reviewed (commented):
I've done refactoring and fixes. They are in separate branch. See #64

mvondracek committed:
mnemoniccli: reraise exceptions, reduces duplicated logic for handling errors

mvondracek committed:
fix: mnemoniccli: type hints

mvondracek commented:
See #64 and #65

sobuch committed:
Merge pull request #64 from mvondracek/Encoding-issues-fix-refactor

Encoding issues fix refactor

sobuch committed:
Merge pull request #65 from mvondracek/mnemoniccli/exceptions

Mnemoniccli/exceptions

lsolodkova committed:
Merge branch 'dev' into Encoding-issues-fix
